src/mscorlib/src/System/Globalization/CompareInfo.cs

```diff
@@ -170,12 +176,20 @@ public static CompareInfo GetCompareInfo(String name)
 
         public static unsafe bool IsSortable(char ch)
         {
+            if (CultureData.InvariantMode)
+            {
+                return true;
+            }
```

Comment:
Rather than having these guards in every function, would it be possible to, for example, implement a special CultureInfo for invariant all in managed code with functions like you have, and then if ICU isn't present / the config switch is thrown, just return that invariant CulutureInfo from anything that can give back a CultureInfo?  Maybe that's not feasible, but it seems like it could help to consolidate the logic better.